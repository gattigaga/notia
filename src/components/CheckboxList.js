import React from "react";
import PropTypes from "prop-types";

import CheckboxInput from "components/CheckboxInput";

const CheckboxList = ({
  items,
  focusIndex,
  isStatic,
  onFocus,
  onClickDelete,
  onChangeLabel,
  onChangeCheck,
  onPressEnter
}) => (
  <div>
    {items.map((item, index) => (
      <CheckboxInput
        key={item.id}
        label={item.label}
        isChecked={item.isChecked}
        isFocus={focusIndex === index}
        isNew={item.isNew}
        isStatic={isStatic}
        onFocus={() => onFocus && onFocus(index)}
        onChangeLabel={e =>
          onChangeLabel && onChangeLabel(e.target.value, item.id)
        }
        onClickDelete={() => onClickDelete && onClickDelete(item.id)}
        onChangeCheck={() => onChangeCheck && onChangeCheck(item.id)}
        onPressEnter={() => onPressEnter && onPressEnter(item.id, index)}
      />
    ))}
  </div>
);

CheckboxList.propTypes = {
  items: PropTypes.array.isRequired,
  focusIndex: PropTypes.number,
  isStatic: PropTypes.bool,
  onFocus: PropTypes.func,
  onClickDelete: PropTypes.func,
  onChangeLabel: PropTypes.func,
  onChangeCheck: PropTypes.func,
  onPressEnter: PropTypes.func
};

CheckboxList.defaultProps = {
  focusIndex: -1
};

export default CheckboxList;
