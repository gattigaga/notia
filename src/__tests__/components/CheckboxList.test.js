import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import CheckboxList from "components/CheckboxList";

let wrapper, items;

beforeEach(() => {
  items = [
    {
      id: 1,
      label: "Learn React",
      isChecked: true,
      isNew: false
    },
    {
      id: 2,
      label: "Learn Redux",
      isChecked: true,
      isNew: false
    },
    {
      id: 3,
      label: "",
      isChecked: false,
      isNew: true
    }
  ];

  wrapper = shallow(<CheckboxList items={items} focusIndex={0} isStatic />);
});

describe("CheckboxList", () => {
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
