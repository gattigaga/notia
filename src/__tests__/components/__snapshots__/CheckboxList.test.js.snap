// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxList renders correctly 1`] = `
<div>
  <CheckboxInput
    isChecked={true}
    isFocus={true}
    isNew={false}
    isStatic={true}
    key="1"
    label="Learn React"
    onChangeCheck={[Function]}
    onChangeLabel={[Function]}
    onClickDelete={[Function]}
    onFocus={[Function]}
    onPressEnter={[Function]}
  />
  <CheckboxInput
    isChecked={true}
    isFocus={false}
    isNew={false}
    isStatic={true}
    key="2"
    label="Learn Redux"
    onChangeCheck={[Function]}
    onChangeLabel={[Function]}
    onClickDelete={[Function]}
    onFocus={[Function]}
    onPressEnter={[Function]}
  />
  <CheckboxInput
    isChecked={false}
    isFocus={false}
    isNew={true}
    isStatic={true}
    key="3"
    label=""
    onChangeCheck={[Function]}
    onChangeLabel={[Function]}
    onClickDelete={[Function]}
    onFocus={[Function]}
    onPressEnter={[Function]}
  />
</div>
`;
